### PR TITLE
Feat support redis v4 and ioredis v5

### PR DESCRIPTION
**Pull request by roggervalf**
this is a breaking change in order to support new version of redis and ioredis
ref https://github.com/animir/node-rate-limiter-flexible/pull/176

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: '3.2'
```
Comment (roggervalf) [inline docker-compose.yml:1]: instead of mocking redis I used a docker instance of redis to test

File: package.json [hunk 2]
```diff
@@ -4,6 +4,8 @@
   "description": "Node.js rate limiter by key and protection from DDoS and Brute-Force attacks in process Memory, Redis, MongoDb, Memcached, MySQL, PostgreSQL, Cluster or PM",
   "main": "index.js",
   "scripts": {
+    "dc:up": "docker-compose -f docker-compose.yml up -d",
```
Comment (roggervalf) [inline package.json:7]: scripts to starts the docker instance

**Comment by animir** [comment]
@roggervalf Wow! This is amazing. I am going to prepare the new release over the weekend. Thank you.

**Comment by animir** [comment]
Released in [3.0.0](https://github.com/animir/node-rate-limiter-flexible/releases/tag/v3.0.0).

**Comment by animir** [comment]
@roggervalf Hi, FYI, I renamed `useIoredis` option to `useRedisPackage`. It was necessary, since I also added `useRedis3AndLowerPackage` flag and kept support of old `redis` packages. Thank you for your contribution.

**Pull request merged**